drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -0,0 +1,1560 @@
+		}
+	}
+
+	for (i = 0; i < NRF_USBD_EPOUT_CNT; i++) {
+		ep_ctx = out_endpoint_ctx(i);
+		__ASSERT_NO_MSG(ep_ctx);
+
+		if (ep_ctx->cfg.en) {
+			nrfx_usbd_ep_enable(ep_addr_to_nrfx(ep_ctx->cfg.addr));
+		}
+	}
+}
+
+static void usbd_handle_state_change(struct nrf_usbd_ctx *ctx)
+{
+	switch (ctx->state) {
+	case USBD_ATTACHED:
+		SYS_LOG_DBG("USB detected");
+		nrfx_usbd_enable();
+		break;
+
+	case USBD_POWERED:
+		SYS_LOG_DBG("USB Powered");
+		ctx->status_code = USB_DC_CONNECTED;
+		ctx->flags |= BIT(NRF_USB_STATUS_CHANGE);
+		usbd_enable_endpoints(ctx);
+		nrfx_usbd_start(true);
+		ctx->ready = true;
+		break;
+
+	case USBD_DETACHED:
+		SYS_LOG_DBG("USB Removed");
+		ctx->ready = false;
+		nrfx_usbd_disable();
+		ctx->status_code = USB_DC_DISCONNECTED;
+		ctx->flags |= BIT(NRF_USB_STATUS_CHANGE);
+		break;
+
+	default:
+		break;
+	}
+
+	if (ctx->flags) {
+		k_work_submit(&ctx->usb_work);
+	}
+}
+
+static void usbd_handle_status_change(struct nrf_usbd_ctx *ctx)
+{
+	if (ctx->status_cb) {
+		ctx->status_cb(ctx->status_code, NULL);
+	}
+}
+
+static inline void usbd_work_process_setup(struct nrf_usbd_ep_ctx *ep_ctx)
+{
+	__ASSERT_NO_MSG(ep_ctx);
+	__ASSERT(ep_ctx->cfg.type == USB_DC_EP_CONTROL,
+		 "Invalid event on CTRL EP.");
+
+	struct usb_setup_packet *usbd_setup;
+
+	/* SETUP packets are handled by USBD hardware.
+	 * For compatibility with the USB stack,
+	 * SETUP packet must be reassembled.
+	 */
+	usbd_setup =
+		(struct usb_setup_packet *)ep_ctx->buf.data;
+	memset(usbd_setup, 0, sizeof(struct usb_setup_packet));
+	usbd_setup->bmRequestType = nrf_usbd_setup_bmrequesttype_get();
+	usbd_setup->bRequest      = nrf_usbd_setup_brequest_get();
+	usbd_setup->wValue        = nrf_usbd_setup_wvalue_get();
+	usbd_setup->wIndex        = nrf_usbd_setup_windex_get();
+	usbd_setup->wLength       = nrf_usbd_setup_wlength_get();
+	ep_ctx->buf.len = sizeof(struct usb_setup_packet);
+
+	SYS_LOG_DBG("SETUP: r:%d rt:%d v:%d i:%d l:%d",
+			(u32_t)usbd_setup->bRequest,
+			(u32_t)usbd_setup->bmRequestType,
+			(u32_t)usbd_setup->wValue,
+			(u32_t)usbd_setup->wIndex,
+			(u32_t)usbd_setup->wLength);
+
+	/* Inform the stack. */
+	ep_ctx->cfg.cb(ep_ctx->cfg.addr, USB_DC_EP_SETUP);
+
+	if (((usbd_setup->bmRequestType & USB_BMREQUESTTYPE_MASK)
+	     == USB_BMREQUESTTYPE_HOSTTODEVICE_MASK)
+	    && (usbd_setup->wLength)) {
+		nrfx_usbd_setup_data_clear();
+	}
+}
+
+static inline void usbd_work_process_recvreq(struct nrf_usbd_ctx *ctx,
+					     struct nrf_usbd_ep_ctx *ep_ctx)
+{
+	if (!ep_ctx->read_pending) {
+		return;
+	}
+	if (!ep_ctx->read_complete) {
+		return;
+	}
+
+	ep_ctx->read_pending = false;
+	ep_ctx->read_complete = false;
+
+	k_sem_take(&ctx->dma_in_use, K_FOREVER);
+	NRFX_USBD_TRANSFER_OUT(transfer, ep_ctx->buf.data,
+			       ep_ctx->cfg.max_sz);
+	nrfx_err_t err = nrfx_usbd_ep_transfer(
+		ep_addr_to_nrfx(ep_ctx->cfg.addr), &transfer);
+	if (err != NRFX_SUCCESS) {
+		SYS_LOG_ERR(
+			"nRF USBD transfer error (OUT): %d.",
+			err);
```

Comment:
You could improve the formatting. Now there is no need to break this line into three pieces. Similar story in [line 572](https://github.com/zephyrproject-rtos/zephyr/pull/8787/files#diff-b6600658d4d8775791cddce8653f308cR572).